core/templates/dev/head/domain/statistics/AnswerDetailsImprovementCardObjectFactorySpec.ts

```diff
@@ -0,0 +1,244 @@
+      'WrittenTranslationObjectFactory',
+      new WrittenTranslationObjectFactory());
+    $provide.value(
+      'WrittenTranslationsObjectFactory',
+      new WrittenTranslationsObjectFactory(
+        new WrittenTranslationObjectFactory()));
+  }));
+  beforeEach(angular.mock.inject(function(
+      _$q_, _$rootScope_, _$uibModal_,
+      _AnswerDetailsImprovementCardObjectFactory_, _ImprovementModalService_,
+      _LearnerAnswerDetailsDataService_,
+      _ANSWER_DETAILS_IMPROVEMENT_CARD_TYPE_, _STATUS_OPEN_) {
+    $q = _$q_;
+    $rootScope = _$rootScope_;
+    $uibModal = _$uibModal_;
+    AnswerDetailsImprovementCardObjectFactory =
+      _AnswerDetailsImprovementCardObjectFactory_;
+    ImprovementModalService = _ImprovementModalService_;
+    LearnerAnswerDetailsDataService = _LearnerAnswerDetailsDataService_;
+    ANSWER_DETAILS_IMPROVEMENT_CARD_TYPE =
+      _ANSWER_DETAILS_IMPROVEMENT_CARD_TYPE_;
+    STATUS_OPEN = _STATUS_OPEN_;
+  }));
+
+  describe('.createNew', function() {
+    it('retrieves data from passed thread', function() {
+      var mockLearnerAnswerDetails = {learnerAnswerInfoData: 'sample'};
+      var card = AnswerDetailsImprovementCardObjectFactory.createNew(
+        mockLearnerAnswerDetails);
+
+      expect(card.getDirectiveData()).toBe(mockLearnerAnswerDetails);
+      expect(card.getDirectiveType()).toEqual(
+        ANSWER_DETAILS_IMPROVEMENT_CARD_TYPE);
+    });
+  });
+
+  describe('.fetchCards', function() {
+    it('fetches threads from the backend', function(done) {
+      spyOn(
+        LearnerAnswerDetailsDataService,
+        'fetchLearnerAnswerInfoData').and.callFake($q.resolve);
+      spyOn(LearnerAnswerDetailsDataService, 'getData').and.returnValue(
+        [{learnerAnswerInfoData: 'abc1'}, {learnerAnswerInfoData: 'def2'}]);
+
+      AnswerDetailsImprovementCardObjectFactory.fetchCards().then(
+        function(cards) {
+          expect(
+            cards[0].getDirectiveData().learnerAnswerInfoData).toEqual('abc1');
+          expect(
+            cards[1].getDirectiveData().learnerAnswerInfoData).toEqual('def2');
+        }).then(done, done.fail);
+
+      // $q Promises need to be forcibly resolved through a JavaScript digest,
+      // which is what $apply helps kick-start.
+      $rootScope.$apply();
+    });
+  });
+
+  describe('AnswerDetailsImprovementCard', function() {
+    beforeEach(function() {
+      this.mockLearnerAnswerDetails = {
+        expId: 12,
+        stateName: 'fakeStateName',
+        learnerAnswerInfoData: [{
+          id: 1,
+          answer: 'fakeAnswer',
+          answer_details: 'fakeAnswerDetails',
+          created_on: 1441870501230.642,
+        }]
+      };
+      this.card =
+        AnswerDetailsImprovementCardObjectFactory.createNew(
+          this.mockLearnerAnswerDetails);
+    });
+
+    describe('.getStatus', function() {
+      it('returns the same status as the thread', function() {
+        expect(this.card.getStatus()).toEqual(STATUS_OPEN);
```

Comment:
Please add an `it` case for `STATUS_NOT_ACTIONABLE`.

Reply:
Done